api/cpp/include/slint.h

```diff
@@ -1049,6 +1049,179 @@ class MapModel : public Model<MappedModelData>
     std::function<MappedModelData(const SourceModelData &)> map_fn;
 };
 
+template<typename ModelData>
+class SortModel;
+
+namespace private_api {
+template<typename ModelData>
+struct SortModelInner : private_api::ModelChangeListener
+{
+    SortModelInner(std::shared_ptr<slint::Model<ModelData>> source_model,
+                   std::function<bool(const ModelData &, const ModelData &)> sort_fn,
+                   slint::SortModel<ModelData> &target_model)
+        : source_model(source_model), sort_fn(sort_fn), target_model(target_model)
+    {
+    }
+
+    void row_added(int first_inserted_row, int count) override
+    {
+        if (sorted_rows_dirty) {
+            reset();
+            return;
+        }
+
+        // Adjust the existing sorted row indices to match the updated source model
+        for (auto it = sorted_rows.begin(), end = sorted_rows.end(); it != end; ++it) {
+            if (*it >= first_inserted_row)
+                *it += count;
+        }
```

Comment:
How about writing this as
```
for (auto it : sorted_rows) {
    
}
```

Reply:
Ah, I think I'm burned from the Qt containers, where the value is not mutable. I'll give it a try :)

Reply:
I've changed this loop now (using `for (auto &row: sorted_rows)` due to the mutation in the center. I looked for other loops to convert, but I couldn't find any. Which ones am I missing? :)